src/pipelines/pipeline.py

```diff
@@ -30,48 +115,36 @@ def __call__(self, text: List[str], **generate_kwargs) -> Tuple[List[str], List[
         num_generated_tokens = [o - i for i, o in zip(input_token_lengths, output_token_lengths)]
 
         output_text = self.tokenizer.batch_decode(output_tokens, skip_special_tokens=True)
+        t3 = time.perf_counter()
+
+        metrics = {
+            NUM_GENERATED_TOKENS: num_generated_tokens,
+            TOKENIZE_TIME: t1 - t0,
+            MODEL_TIME: t2 - t1,
+            DECODE_TIME: t3 - t2,
+            END_TO_END_TIME: t3 - t0,
+        }
 
-        return output_text, num_generated_tokens
+        return output_text, metrics
 
     def get_num_parameters(self) -> int:
-        param_count = 0
-        for i in self.model.parameters():
-            param_count += i.numel()
-        return param_count
-
-
-def get_config_tokenizer_model_class(args: Namespace) -> Union[BloomConfig, GPT2Config]:
-    tokenizer = AutoTokenizer.from_pretrained("gpt2")
-    tokenizer.add_special_tokens({"pad_token": "[PAD]"})
-
-    if args.model_class.lower() == "bloom":
-        config = BloomConfig(
-            attention_softmax_in_fp32=True,
-            hidden_size=args.hidden_size,
-            n_head=args.n_head,
-            n_layer=args.n_layer,
-            vocab_size=len(tokenizer),
-            bos_token_id=tokenizer.bos_token_id,
-            eos_token_id=tokenizer.eos_token_id,
-            use_cache=True,
-        )
-        model_class = BloomForCausalLM
-    elif args.model_class.lower() == "gpt2":
-        config = GPT2Config(
-            n_embd=args.hidden_size,
-            n_head=args.n_head,
-            n_layer=args.n_layer,
-            n_positions=args.n_positions,
-            bos_token_id=tokenizer.bos_token_id,
-            eos_token_id=tokenizer.eos_token_id,
-            attention_type=args.attention_type,
-            print_details=False,
-            vocab_size=len(tokenizer),
-            use_cache=True,
-        )
-        model_class = GPT2LMHeadModel
-
-    if not os.path.exists("tmp"):
-        model_class._from_config(config).save_pretrained("tmp")
-
-    return config, tokenizer, model_class
+        return sum(p.numel() for p in self.model.parameters())
+
+    def aggregate_and_format_metrics(self, metrics: List[Dict[str, Any]]):
```

Comment:
@mayank31398 It's better to leave this encapsulated with the class that generates these metrics.